### ISSUE TITLE
(orignal theme compatibility patch) Do not color arithmetical operators

### ISSUE THREAD

**Issue by Trismegistos84** [SublimeColors/Solarized#37]
Solarized Dark and Light themes orginally does not color aritchmetical operators e.g. (=, +, -, ++, +=, &&, ||) for any language e.g.

python ![python](http://ethanschoonover.com/solarized/img/screen-python-dark.png)

java ![java](http://ethanschoonover.com/solarized/img/screen-java-dark.png)

Sublime Text 3 port follows these setting in some languages e.g. C++, Haskell but for other languages like Python or Ruby it colors these operators same as keywords. This is deviation from orginal theme and additionally it introduces noise.

I have prepared lines that has to be added to make Sublime scheme compatible with orginal one.
### Add following lines to dark scheme:

```xml
<dict>
    <key>name</key>
    <string>Arithmetical, Assignment, Comparision Operators</string>
    <key>scope</key>
    <string>keyword.operator.comparison, keyword.operator.assignment, keyword.operator.arithmetic</string>
    <key>settings</key>
    <dict>
        <key>fontStyle</key>
        <string></string>
        <key>foreground</key>
        <string>#839496</string>
    </dict>
</dict>
```

### Add following lines to light scheme:

```xml
<dict>
    <key>name</key>
    <string>Arithmetical, Assignment, Comparision Operators</string>
    <key>scope</key>
    <string>keyword.operator.comparison, keyword.operator.assignment, keyword.operator.arithmetic</string>
    <key>settings</key>
    <dict>
        <key>fontStyle</key>
        <string></string>
        <key>foreground</key>
        <string>#657B83</string>
    </dict>
</dict>
```

**Comment by braver** [comment]
This implementation of solarized strives for internal consistency over compliance with the "original". Your arguments make sense though and the changes make sense. If you can put it in a pull request and provide before and after screenshots, I'll happily pull in the changes.

**Comment by Trismegistos84** [comment]
I have created pull request with elaborate explanation https://github.com/SublimeColors/Solarized/pull/38.

**Comment by Trismegistos84** [comment]
Thank *you*

**Comment by jibsen** [comment]
Just to add a little context, I think this is another of those issues where you could debate the intent of the original Vim highlighting, or simply follow how it looks.

If you look at the [relevant section of `solarized.vim`](https://github.com/altercation/vim-colors-solarized/blob/master/colors/solarized.vim#L561), you can see that operators are listed as a subgroup of statements. The groups listed in the comments come from the Vim documentation (`:help group-name`).

But the syntax files provided with Vim are relatively simple, and do not allow highlighting operators in the languages I checked (there are no rules for them).

Personally, I think either works fine. I can see the argument for removing unnecessary noise.

**Comment by braver** [comment]
Yes, you're right. We shouldn't let limitations of Vim dictate the syntax highlighting in Sublime if the latter has a much richer palette. I think I would've pulled this if it added colors to all syntaxes as well instead of removing it everywhere. As you say, either way is fine, but consistency across languages in the same editor is a nice thing to have.
